src/phase0/state_transition/slot_processing.rs

```diff
@@ -0,0 +1,77 @@
+use crate::phase0::beacon_state::BeaconState;
+use crate::phase0::mainnet::Error::InvalidSlot;
+use crate::phase0::state_transition::{Context, Error};
+use crate::phase0::state_transition::epoch_processing::process_epoch;
+use crate::primitives::{Slot, Bytes32};
+use ssz_rs::prelude::*;
+
+pub fn process_slots<
+    const SLOTS_PER_HISTORICAL_ROOT: usize,
+    const HISTORICAL_ROOTS_LIMIT: usize,
+    const ETH1_DATA_VOTES_BOUND: usize,
+    const VALIDATOR_REGISTRY_LIMIT: usize,
+    const EPOCHS_PER_HISTORICAL_VECTOR: usize,
+    const EPOCHS_PER_SLASHINGS_VECTOR: usize,
+    const MAX_VALIDATORS_PER_COMMITTEE: usize,
+    const PENDING_ATTESTATIONS_BOUND: usize,
+>(
+    state: &mut BeaconState<
+        SLOTS_PER_HISTORICAL_ROOT,
+        HISTORICAL_ROOTS_LIMIT,
+        ETH1_DATA_VOTES_BOUND,
+        VALIDATOR_REGISTRY_LIMIT,
+        EPOCHS_PER_HISTORICAL_VECTOR,
+        EPOCHS_PER_SLASHINGS_VECTOR,
+        MAX_VALIDATORS_PER_COMMITTEE,
+        PENDING_ATTESTATIONS_BOUND,
+    >,
+    slot: &Slot,
```

Comment:
no need to pass this by ref here, can just use `Slot` directly

Reply:
ok